src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -62,23 +89,16 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 				msg.getRequestHeader().isSecure()){ //No point reporting for non-SSL resources
 			//Content available via both HTTPS and HTTP is a separate though related issue
 			Vector<String> STSOption = msg.getResponseHeader().getHeaders("Strict-Transport-Security");
-			if (STSOption == null) { // Header NOT found
-				Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_MEDIUM, //PluginID, Risk, Reliability
-						getName()); //Name
-			    		alert.setDetail(
-			    			getDescription(), //Description
-			    			msg.getRequestHeader().getURI().toString(), //URI
-			    			"",	// Param
-			    			"", // Attack
-			    			"", // Other info
-			    			getSolution(), //Solution
-			    			getReference(), //References
-			    			"",	// Evidence
-							16, // CWE-16: Configuration
-							15,	//WASC-15: Application Misconfiguration
-			    			msg); //HttpMessage
-			    		parent.raiseAlert(id, alert);
+			if (STSOption == null && msg.getRequestHeader().isSecure()) { // Header NOT found
+					raiseAlert(VULN_TYPE.HSTS_MISSING, msg, id);
+					return; //No point continuing
+			} else { 
+				for (String stsHeader : STSOption) {

```

Comment:
Exactly, that's why I think we should alert about that (1st case), lets say that the application is setting the header but other layer is also setting it [1]... in this case the use of HTTPS will not be enforced (RFC says that the first header should be processed, but in the end it really depends on the user agent's implementation, Firefox seems to reject this case).

Another case (similar to 3rd) is when the header is present but its contents are invalid [2], this must/should be rejected by UAs... so, despite the good intentions, it will have no effect.

Anyway, this is unrelated to changes proposed in the pull request. Should we go ahead with the merge?

[1] Example with two headers:
```
Server: Microsoft-IIS/8.5
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Strict-Transport-Security: max-age=31536000; includeSubDomains
x-content-type-options: nosniff
```
[2] Example of malformed header:
`Strict-Transport-Security: “max-age=31536000″`